File: go/codewars/7kyu/sum_of_odd_numbers.go
package main

import "fmt"

// RowSumOddNumbers returns the sum of odd numbers in a row
//
//	    		  1
//			   3     5
//			7     9    11
//		13    15    17    19
//	  21    23    25    27    29
//
// n =1 -> 1
// n =2 -> 3 + 5 = 8
func RowSumOddNumbers(n int) int {
	return n * n * n
}

func main() {
	fmt.Println(RowSumOddNumbers(1) == 1)
	fmt.Println(RowSumOddNumbers(2) == 8)
	fmt.Println(RowSumOddNumbers(3) == 27)
	fmt.Println(RowSumOddNumbers(4) == 64)
	fmt.Println(RowSumOddNumbers(5) == 125)
}

File: ruby/problem_solving/codewars/7kyu/sum_of_odd_numbers.rb
def row_sum_odd_numbers(n)
  start_number = n * n - n + 1
  (0...n).map { |i| start_number + 2 * i }.sum
end

p row_sum_odd_numbers(1) == 1
p row_sum_odd_numbers(2) == 8
p row_sum_odd_numbers(13) == 2197
p row_sum_odd_numbers(19) == 6859
p row_sum_odd_numbers(41) == 68921